apps/server/src/modules/tool/external-tool/uc/external-tool.uc.ts

```diff
@@ -88,8 +91,38 @@ export class ExternalToolUc {
 		return metadata;
 	}
 
-	private async ensurePermission(userId: EntityId, permission: Permission) {
+	private async ensurePermission(userId: EntityId, permission: Permission): Promise<void> {
 		const user: User = await this.authorizationService.getUserWithPermissions(userId);
 		this.authorizationService.checkAllPermissions(user, [permission]);
 	}
+
+	public async getDatasheet(userId: EntityId, externalToolId: EntityId): Promise<Buffer> {
+		const user: User = await this.authorizationService.getUserWithPermissions(userId);
+		this.authorizationService.checkAllPermissions(user, [Permission.TOOL_ADMIN]);
+		const dataSheetData: ExternalToolDatasheetTemplateData =
+			await this.externalToolService.getExternalToolDatasheetTemplateData(
+				externalToolId,
+				user.firstName,
+				user.lastName
+			);
+		const buffer: Promise<Buffer> = firstValueFrom(
```

Comment:
Cedric would probably like it if you made an extra variable for the observable before using firstValueFrom